chartist-plugin-legend.js

```diff
@@ -82,23 +100,36 @@
                 legendNames = chart.data.labels;
             }
             legendNames = options.legendNames || legendNames;
-            
+
             // Check if given class names are viable to append to legends
             var classNamesViable = (Array.isArray(options.classNames) && (options.classNames.length === legendNames.length));
-            
+
             // Loop through all legends to set each name in a list item.
-            legendNames.forEach(function (legend, i) {
-                var li = document.createElement('li');
-                li.className = 'ct-series-' + i;
-                // Append specific class to a legend element, if viable classes are given
-                if (classNamesViable) {
-                   li.className += ' ' + options.classNames[i];
-                }
-                li.setAttribute('data-legend', i);
-                li.textContent = legend.name || legend;
-                legendElement.appendChild(li);
+            for (var legendNamesIndex = 0, length = legendNames.length; legendNamesIndex < length; legendNamesIndex++) {
```

Comment:
If jsperf wasn't down you could see the reference test [http://jsperf.com/for-vs-foreach/66](here). Let's hope jsperf v2 comes soon.